src/elements/ifc-client-link.ts

```diff
@@ -0,0 +1,40 @@
+import { Client } from '../client';
+const PATH_ATTR = 'path';
+
+/**
+ * A custom element responsible for rendering an anchor element, turning a path
+ * within the current application to a full top-level url.
+ * It will be registered as `ifc-client-link` by a call to [[registerCustomElementsClient]] and
+ * should not be created directly. Instead add a `<ifc-client-link>` element to your
+ * markup or use `document.createElement('ifc-client-link')` after calling
+ * [[registerCustomElementsClient]].
+ */
+
+export default class IfcClientLinkElement extends HTMLElement {
+  private _client: Client;
+
+  /** @internal */
+  constructor(clientInstance:any) {
+    super();
+    this._client = clientInstance;
+  }
+
+  /**
+   * @internal
+   * @inheritdoc
+   */
+  static get observedAttributes() {
+    return [PATH_ATTR];
+  }
+
+  /**
+   * @internal
+   * @inheritdoc
+   */
+  public connectedCallback() {
+    const thisInnerHTML = this.innerHTML;
+    let path = this.getAttribute(PATH_ATTR) || 'about:blank';
+    path = this._client.urlFromClientPath(path);
+    this.innerHTML = `<a href=${path}>${thisInnerHTML}</a>`;
```

Comment:
Rather than setting `innerHTML` in the `connectedCallback`, I think we should be using the shadow DOM and regular DOM APIs or a template to build up the structure.  I think that's normally done in the constructor, but it could be done here as well.

There are some good examples from [google](https://developers.google.com/web/fundamentals/web-components/customelements#shadowdom) and [mozilla](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM) that should help with getting started.

Reply:
I was able to solve the above comment by adding the `target=_top` attribute to the link and adding the `allow-top-navigation-by-user-activation` keyword to the iframe sandbox